### DISCUSSION TITLE
IBM i Actions

### DISCUSSION THREAD

**Discussion by jkdavew** [codefori/vscode-ibmi#1293] [Ideas]
What are thoughts on eliminating parameters with literals listed in the out of the box IBM i Actions.  Leaving them out would allow the system defined command defaults to be used when the action is ran.

Example 1: Currently
CRTCMD CMD(&OPENLIB/&OPENMBR) PGM(&OPENLIB/&OPENMBR) SRCFILE(&OPENLIB/&OPENSPF) ALLOW(*ALL) CURLIB(*NOCHG) PRDLIB(*NOCHG)

Example 1: Future
CRTCMD CMD(&OPENLIB/&OPENMBR) PGM(&OPENLIB/&OPENMBR) SRCFILE(&OPENLIB/&OPENSPF)

Example 2: Currently
CRTSRVPGM SRVPGM(&OPENLIB/&OPENMBR) EXPORT(*ALL) BNDSRVPGM(*NONE) BNDDIR(*NONE) ACTGRP(*CALLER) TGTRLS(*CURRENT) 

Example 2: Future
CRTSRVPGM SRVPGM(&OPENLIB/&OPENMBR)

**Comment by worksofliam** [comment]
Next best thing is a PR so it can be reviewed by many people :)

**Comment by Gregory1701** [comment]
This recently bit me.  Having EXPORT(*ALL) on CRTSRVPGM command caught me off guard.  In my shop, the system default is EXPORT(*SRCFILE).  I think it's a better approach, to leave out the literal parameters and default to the users' system default.
